### PR TITLE
feat:  support the operator make use of  the config-reloader tls and basic authentication with crds

### PR DESCRIPTION
**Pull request by dongjiang1989**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

ref: https://github.com/prometheus-operator/prometheus-operator/pull/5932#issuecomment-1761545280

Support the operator make use of  the config-reloader tls and basic authentication with crds 

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [x] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Support the operator make use of  the config-reloader tls and basic authentication with crds
```

### CODE REVIEW

**Comment by dongjiang1989** [comment]
https://github.com/prometheus-operator/prometheus-operator/pull/5932#issuecomment-1761545280

@simonpasquier  PTAL

File: pkg/prometheus/agent/statefulset.go [hunk 1]
```diff
@@ -236,6 +255,24 @@ func makeStatefulSetSpec(
 		webConfigGenerator.Warn("web.config.file")
 	}
 
+	// Mount web config and web TLS credentials as volumes for config-reloader.
+	// We always mount the web config file for versions greater than 0.69.0.
+	if cfgReloaderVersion.GTE(semver.MustParse("0.69.0")) && cpf.WebConfigReloader != nil {
```
Comment (nicolastakashi) [inline pkg/prometheus/agent/statefulset.go:260]: Just wondering if we can abstract this code in a func and reuse between agent and server.

Comment (dongjiang1989) [reply pkg/prometheus/agent/statefulset.go:260]: Thank you for your review. But abstracting this code into a func will have too many parameters. Probably not worth it 🤔️

**Review by simonpasquier — COMMENTED** [review]
I'm wondering what's the correct API for this feature. Adding a `.spec.webConfigReloader` field increases the API surface and I'm not 100% sure that it's required. Maybe it's good enough to say "reuse the Prometheus web config for the reloader"?

**Comment by dongjiang1989** [comment]
> I'm wondering what's the correct API for this feature. Adding a `.spec.webConfigReloader` field increases the API surface and I'm not 100% sure that it's required. Maybe it's good enough to say "reuse the Prometheus web config for the reloader"?

Thanks for your review. 
Adding`.spec.webConfigReloader` allows for more flexible configuration of config-reloader. 
But it can also be reused the Prometheus/AlertManager web config and add a `.spec.enableWebConfigReloader` flag field to control whether to enable this function.

Which method do you think is most suitable?

File: pkg/prometheus/agent/statefulset.go (hunk 1 above)
Comment (nicolastakashi) [reply pkg/prometheus/agent/statefulset.go:260]: I'm kinda Ok having some parameters on the func signature.
If we think it's to much we can always create an struct and send it to the func.

WDYT?

**Comment by dongjiang1989** [comment]
Closing. 
Add the operator crds to make use of config-reloader web-config-file : https://github.com/prometheus-operator/prometheus-operator/pull/6194

**Pull request closed without merging**